File: opspilot-enterprise/packages/shared-schema/src/opspilot_schema/upgrade.py
from __future__ import annotations
from typing import Optional
from pydantic import BaseModel


class UpgradePackage(BaseModel):
    id: str
    version: str
    release_name: str
    description: str
    changelog: list[str] = []
    status: str
    target: str
    package_size_mb: float
    requires_restart: bool
    requires_approval: bool
    risk_level: str
    released_at: str
    deployed_at: Optional[str] = None
    deployed_by: Optional[str] = None
    rollback_version: Optional[str] = None
    environment: str


class UpgradeDeploymentRecord(BaseModel):
    id: str
    package_id: str
    package_version: str
    status: str
    environment: str
    deployed_by: str
    started_at: str
    completed_at: Optional[str] = None
    log_summary: list[str] = []
    rollback_available: bool = False

File: opspilot-enterprise/packages/shared-types/src/upgrade.ts
export type UpgradeStatus = "available" | "downloading" | "ready" | "deploying" | "deployed" | "failed" | "rolled_back";
export type UpgradeTarget = "web-frontend" | "api-bff" | "orchestrator" | "tool-gateway" | "vmware-skill-gateway" | "all-services";

export interface UpgradePackage {
  id: string;
  version: string;
  release_name: string;
  description: string;
  changelog: string[];
  status: UpgradeStatus;
  target: UpgradeTarget;
  package_size_mb: number;
  requires_restart: boolean;
  requires_approval: boolean;
  risk_level: string;
  released_at: string;
  deployed_at: string | null;
  deployed_by: string | null;
  rollback_version: string | null;
  environment: "dev" | "staging" | "production";
}

export interface UpgradeDeploymentRecord {
  id: string;
  package_id: string;
  package_version: string;
  status: UpgradeStatus;
  environment: string;
  deployed_by: string;
  started_at: string;
  completed_at: string | null;
  log_summary: string[];
  rollback_available: boolean;
}
